src/libpriv/rpmostree-container.cxx

```diff
@@ -60,3 +60,16 @@ rpmostree_container_rebuild (rpmostreecxx::Treefile &treefile,
 
   return TRUE;
 }
+
+gboolean
+rpmostree_container_install_packages (char **packages,
+                                      GCancellable  *cancellable,
+                                      GError       **error)
+{
+  // XXX: awkward: add a e.g. `treefile_new_client_from_fields()`
+  g_autofree char *joined = g_strjoinv ("\",\"", packages);
+  g_autofree char *treefile_s = g_strdup_printf ("{\"packages\": [\"%s\"]}", joined);
```

Comment:
Eeek.  I think today we can do `treefile_new_empty()` and then let's add a `set_packages()` API?

Reply:
Let's keep that in a follow-up? I'd like to consider something more generic than per-field methods.